### PR TITLE
add initial markdown credits implementation (#3326)

### PR DESCRIPTION
**Pull request by asiekierka**
### Contains

Initial implementation for parsing and viewing Credits.md. Involved adding word wrapping functionality to UIScrollingText and some parsing in CreditsScreen. (#3326)

### How to test

1. Copy docs/Credits.md to engine/src/main/resources/
2. Start game
3. Press "Credits" button

### Outstanding before merging

- [ ] Add Gradle code to copy docs/Credits.md to engine/src/main/resources
- [ ] Figure out why ~~a few places have an unnecessary extra line separator~~ a few lines seemingly get omitted and replaced with blanks

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by Cervator** [comment]
@GooeyHub: add to whitelist

@asiekierka: heya, thanks for that! :-) Will check out when I get a chance

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/CreditsScreen.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/CreditsScreen.java]: I guess this was used early on, but now is unused - remove? :-)

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/CreditsScreen.java [hunk 2]
```diff
@@ -50,15 +51,28 @@ public void initialise() {
         creditsScroll = find("creditsScroll", UIScrollingText.class);
         if (creditsScroll != null) {
             ClassLoader classloader = getClass().getClassLoader();
-            InputStream is = classloader.getResourceAsStream("Credits.txt");
+            InputStream is = classloader.getResourceAsStream("Credits.md");
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/CreditsScreen.java]: Could you add a check for this resulting in a valid file? And if not then just show a simple warning message on the credits screen. Maybe something about it possibly being due to running from source where credits are unavailable

On a related note could you please throw an entry into the `.gitignore` to cover where this file would end up?

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Thanks for the tweaks :-)

I added the Gradle bit and another tweak myself as the translation system isn't quite that automagic - but the addition is pretty trivial. Check it out at https://github.com/MovingBlocks/Terasology/commit/ef4655b66c924de52513a6a37b4158dc61fa90cd

Also formally added you to the credits 👍

**Pull request merged**